root/client/src/util/tablehelpers.js

```diff
@@ -51,4 +51,91 @@ export const createPossibleTimeBlock = (startTime, duration) => {
   span.innerHTML = timeframe;
   span.classList.add('temp-timeblock', position, `h-${duration}`);
   return span;
+}
+
```

Comment:
this is a good first implementation of a dictionary for finding dates. i think this is fine for this PR, but i do think there might be a better way to calculate all of this, would just need to think on it, but that can be scoped out as an optimization ticket afterwards, no need to stress about changing it now if it works.

Reply:
Sounds good, i'll mark this as resolved and saved this for future purposes.